src/plotting/plot.component.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+  Chart as ChartJS,
+  ChartData,
+  ChartOptions,
+  Legend,
+  LinearScale,
+  LogarithmicScale,
+  LineElement,
+  PointElement,
+  TimeScale,
+  Title,
+  Tooltip,
+  ScatterController,
+  LineController,
+} from 'chart.js';
+import 'chartjs-adapter-date-fns';
+import React from 'react';
+import { Chart } from 'react-chartjs-2';
+import { useRecords } from '../api/records';
+import { AxisSettings, PlotType } from '../app.types';
+
+ChartJS.register(
+  LinearScale,
+  LogarithmicScale,
+  TimeScale,
+  PointElement,
+  LineElement,
+  Tooltip,
+  Legend,
+  Title,
+  ScatterController,
+  LineController
+);
+
+type FullPlotProps = {
+  data: ChartData<PlotType>;
+} & PlotProps;
+
+export const Plot = (props: FullPlotProps) => {
+  const { data, title, type, XAxisSettings, YAxesSettings } = props;
+
+  const options = React.useMemo(() => {
+    const options: ChartOptions<PlotType> = {
+      plugins: {
+        title: {
+          text: title,
+          display: true,
+        },
+      },
+      scales: {
+        x: {
+          type: XAxisSettings.scale,
+        },
+        y: {
+          type: YAxesSettings.scale,
+        },
+      },
+    };
+    return options;
+  }, [title, XAxisSettings, YAxesSettings]);
+
+  return (
+    <Chart
+      data={data}
+      options={options}
+      type={type}
+      aria-label={`${title} plot`}
+    />
+  );
+};
+
+interface PlotProps {
+  title: string;
+  type: PlotType;
+  XAxisSettings: AxisSettings;
+  YAxesSettings: AxisSettings;
```

Comment:
Minor typo, change to `YAxisSettings`

Reply:
This isn't a typo ;) There's potentially two Y axes (it's in a later requirement), hence using the plural axes

Reply:
Oh yes, you're right, my bad. I assume they'd have the same settings applied to both? My mind is still trying to get around two y-axes

Reply:
The two axes can have independent settings - not sure how they want this to work in the UI but it's definitely possible using chart.js: https://react-chartjs-2.js.org/examples/multiaxis-line-chart/